crates/apub/src/activities/mod.rs

```diff
@@ -256,13 +256,13 @@ pub async fn match_outgoing_activities(
         )
         .await
       }
-      RemovePost(post, person, data) => {
+      RemovePost(post, person, reason, removed) => {
         send_apub_delete_in_community_new(
           person,
           post.community_id,
           DeletableObjects::Post(post.into()),
-          data.reason.or_else(|| Some(String::new())),
-          data.removed,
+          reason.or_else(|| Some(String::new())),
```

Comment:
This can be an option tho right? I'm guessing this is for compatibility reasons?

Reply:
From docs of the method:

> Parameter `reason` being set indicates that this is a removal by a mod. If its unset, this action was done by a normal user.

Its a bit hacky but it works.